pkg/providers/instancetype/suite_test.go

```diff
@@ -759,13 +759,68 @@ var _ = Describe("InstanceType Provider", func() {
 		})
 	})
 
+	Context("ImageProvider + Image Family", func() {
+		DescribeTable("should select the right image for a given instance type",
+			func(instanceType string, imageFamily string, expectedImageDefinition string, expectedGalleryURL string) {
+				nC := test.AKSNodeClass()
+				nC.Spec.ImageFamily = lo.ToPtr(imageFamily)
+				coretest.ReplaceRequirements(nodePool, v1.NodeSelectorRequirement{
+					Key:      v1.LabelInstanceTypeStable,
+					Operator: v1.NodeSelectorOpIn,
+					Values:   []string{instanceType},
+				})
+				nodePool.Spec.Template.Spec.NodeClassRef = &corev1beta1.NodeClassReference{Name: nC.Name}
+				ExpectApplied(ctx, env.Client, nodePool, nC)
+				pod := coretest.UnschedulablePod(coretest.PodOptions{})
+				ExpectProvisioned(ctx, env.Client, cluster, cloudProvider, coreProvisioner, pod)
+				ExpectScheduled(ctx, env.Client, pod)
+
+				Expect(azureEnv.VirtualMachinesAPI.VirtualMachineCreateOrUpdateBehavior.CalledWithInput.Len()).To(Equal(1))
+				vm := azureEnv.VirtualMachinesAPI.VirtualMachineCreateOrUpdateBehavior.CalledWithInput.Pop().VM
+				Expect(vm.Properties.StorageProfile.ImageReference).ToNot(BeNil())
+				Expect(vm.Properties.StorageProfile.ImageReference.CommunityGalleryImageID).ToNot(BeNil())
+				parts := strings.Split(*vm.Properties.StorageProfile.ImageReference.CommunityGalleryImageID, "/")
+				Expect(parts[2]).To(Equal(expectedGalleryURL))
+				Expect(parts[4]).To(Equal(expectedImageDefinition))
+
+				// Need to reset env since we are doing these nested tests
+				cluster.Reset()
+				azureEnv.Reset()
+			},
+			Entry("Gen2, Gen1 instance type with AKSUbuntu image family",
+				"Standard_D2_v5", "Ubuntu2204", "2204gen2containerd", v1alpha2.AKSUbuntuPublicGalleryURL),
```

Comment:
Can these use the constants like `v1alpha2.Ubuntu2204ImageFamily` etc.?

Reply:
yes that works for ubuntu2204, not for 2204gen2containerd

Reply:
Why not?

Reply:
I was thinking we might want to reference the imagefamily os and such in instance provider at some point but I don't think this is the case. i can move the consts back to image family

Reply:
changed back